### PR TITLE
ENH code to apodize image edges

### PR DESCRIPTION
**Pull request by beckermr**
This code apodizes image edges. 

cc @esheldon

### CODE REVIEW

**Comment by beckermr** [comment]
This works great. 

Here is an image with an artifact

bad

And here is the same image with anodization applied before metacal:

good

notebook with this example is here: https://gist.github.com/beckermr/d0f437099e1be0265326ac19898e150c

File: metadetect/masking.py [hunk 1]
```diff
@@ -3,6 +3,55 @@
...
+    mask_bit_val: int
+        The bit to set in the bit mask for areas that are apodized.
+    ap_rad: float
+        When apodizing, the scale of the kernel. The total kernel goes from 0 to 1
+        over 6*ap_rad.
+    """
+    ap_mask = np.ones_like(mbobs[0][0].image)
+    _build_square_apodization_mask(ap_rad, ap_mask)
+
+    msk = ap_mask < 1
```
Comment (esheldon) [inline metadetect/masking.py:23]: Using a bool mask `ap_mask < ` means you need to check it at every pixel, every time it is used.  when you do `image[msk]` it has to check "is msk True?" at every pixel.  In this case a small fraction of the pixels are going to have `ap_mask < 1` so its faster to do
```python
msk = np.where(ap_mask < 1)
if msk[0].size > 0:
    for obslist in mbobs
        ...
        obs.image[msk] *= ap_mask[msk]
```
later you do `np.all(msk)`, again hitting every pixel, but with above you can again check `msk[0].size == image.size`.

Maybe this optimization doesn't matter, maybe it does.

**Review by esheldon — COMMENTED** [review]
This is great, I'm interested to try it. I have one comment about performance.  Maybe it doesn't matter when we are doing FFTs but might be worth thinking about the alternate pattern

File: metadetect/masking.py (hunk 1 above)
Comment (beckermr) [reply metadetect/masking.py]: Good points! Fixed!

File: metadetect/masking.py [hunk 2]
```diff
@@ -3,6 +3,55 @@
...
+    ap_mask = np.ones_like(mbobs[0][0].image)
+    _build_square_apodization_mask(ap_rad, ap_mask)
+
+    msk = np.where(ap_mask < 1)
+    if msk[0].size > 0:
+        for obslist in mbobs:
+            for obs in obslist:
+                # the pixels list will be reset upon exiting
+                with obs.writeable():
+                    obs.image *= ap_mask
```
Comment (esheldon) [inline metadetect/masking.py:29]: this could also be obs.image[msk] *= ap_msk[msk], but I understand if you want to move on

**Review by esheldon — APPROVED** [review]

File: metadetect/masking.py (hunk 2 above)
Comment (beckermr) [reply metadetect/masking.py]: done!

**Pull request merged**